test/extended/openstack/servergroup.go

```diff
@@ -6,80 +6,114 @@ import (
-		o.Expect(err).NotTo(o.HaveOccurred())
+	oc = exutil.NewCLI("openstack")
 
-		g.By("getting the IDs of the Control plane instances and their Server group")
-		masterInstanceUUIDs := make([]interface{}, 0, 3)
-		var serverGroupName string
-		{
+	// OCP 4.5: https://issues.redhat.com/browse/OSASINFRA-1300
+	g.Context("when installed", func() {
+		g.BeforeEach(func() {
+			ctx = context.TODO()
 
-			clientSet, err := e2e.LoadClientset()
+			g.By("preparing a dynamic client")
+			cfg, err := e2e.LoadConfig()
 			o.Expect(err).NotTo(o.HaveOccurred())
-
-			nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{
-				LabelSelector: "node-role.kubernetes.io/master",
-			})
+			dc, err = dynamic.NewForConfig(cfg)
 			o.Expect(err).NotTo(o.HaveOccurred())
 
-			ms := dc.Resource(schema.GroupVersionResource{
-				Group:    "machine.openshift.io",
-				Resource: "machines",
-				Version:  "v1beta1",
-			})
-			for _, item := range nodeList.Items {
-				uuid := strings.TrimPrefix(item.Spec.ProviderID, "openstack:///")
-				masterInstanceUUIDs = append(masterInstanceUUIDs, uuid)
+			g.By("getting the IDs of the Control plane instances and their Server group")
+			masterInstanceUUIDs := make([]interface{}, 0, 3)
+			{
+				clientSet, err := e2e.LoadClientset()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+					LabelSelector: "node-role.kubernetes.io/master",
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
 
-				machineAnnotation := strings.SplitN(item.Annotations["machine.openshift.io/machine"], "/", 2)
-				o.Expect(machineAnnotation).To(o.HaveLen(2))
+				ms := dc.Resource(schema.GroupVersionResource{
+					Group:    "machine.openshift.io",
+					Resource: "machines",
+					Version:  "v1beta1",
+				})
+				for _, item := range nodeList.Items {
+					uuid := strings.TrimPrefix(item.Spec.ProviderID, "openstack:///")
+					masterInstanceUUIDs = append(masterInstanceUUIDs, uuid)
+
+					machineAnnotation := strings.SplitN(item.Annotations["machine.openshift.io/machine"], "/", 2)
+					o.Expect(machineAnnotation).To(o.HaveLen(2))
+
+					res, err := ms.Namespace(machineAnnotation[0]).Get(ctx, machineAnnotation[1], metav1.GetOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					instanceServerGroupNameField := getFromUnstructured(res, "spec", "providerSpec", "value", "serverGroupName")
+					o.Expect(instanceServerGroupNameField).NotTo(o.BeNil(), "the Server group name should be present in the Machine definition")
+
+					instanceServerGroupName := instanceServerGroupNameField.(string)
+					o.Expect(instanceServerGroupName).NotTo(o.BeEmpty(), "the Server group name should not be the empty string")
+					if serverGroupName == "" {
+						serverGroupName = instanceServerGroupName
+					} else {
+						o.Expect(instanceServerGroupName).To(o.Equal(serverGroupName), "two Control plane Machines have different serverGroupName set")
+					}
```

Comment:
This block is really part of the test below, and should not be executed in a before" stanza

Reply:
Done